src/components/elements/table/RestockPurchasesTable.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {useEffect} from 'react';
+
+import Paper from '@material-ui/core/Paper';
+import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TablePagination from '@material-ui/core/TablePagination';
+import TableRow from '@material-ui/core/TableRow';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+
+import {RestockDataEntry} from '../../../state/restock/restockData';
+import {RestockStatusType} from '../../../state/restock/restockData';
+import {restockDispatchers} from '../../../state/restock/restockDispatchers';
+import {useRestockSelector} from '../../../state/restock/restockSelector';
+import {useDispatch} from '../../../state/store';
+import {stableSort, getComparator, Order} from '../../../utils/Sort';
+import {NoData} from './NoData';
+import PurchaseModal from './PurchaseModal';
+import Status from './Status';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: '100%',
+    },
+    paper: {
+      width: '100%',
+      marginBottom: theme.spacing(2),
+    },
+    table: {
+      minWidth: 750,
+    },
+    visuallyHidden: {
+      border: 0,
+      clip: 'rect(0 0 0 0)',
+      height: 1,
+      margin: -1,
+      overflow: 'hidden',
+      padding: 0,
+      position: 'absolute',
+      top: 20,
+      width: 1,
+    },
+    tableRow: {
+      backgroundColor: '#e3e3e3',
+    },
+    tablePagination: {
+      backgroundColor: '#e3e3e3',
+    },
+  }),
+);
+
+const statusColors: { [key in RestockStatusType]: any } = {
+  completed: 'success',
+  processing: 'info',
+  shipped: 'danger',
+};
+
+type HeadCell = {
+  columnName: keyof RestockDataEntry;
+  label: string;
+  numeric: boolean;
+}
+
+const headCells: Array<HeadCell> = [
+  {columnName: 'id', numeric: false, label: 'Order #'},
+  {columnName: 'purchaseDate', numeric: false, label: 'Purchase Date'},
+  {columnName: 'status', numeric: false, label: 'Status'},
+  {columnName: 'type', numeric: false, label: 'Type'},
+  {columnName: 'totalPrice', numeric: true, label: 'Total Price ($)'},
+];
+
+type EnhancedTableProps = {
+  classes: ReturnType<typeof useStyles>;
+  onRequestSort: (event: React.MouseEvent<HTMLSpanElement>, property: keyof RestockDataEntry) => void;
+  order: Order;
+  orderBy: string;
+}
+
+const EnhancedTableHead = (props: EnhancedTableProps) => {
```

Comment:
Move to separate file

Reply:
I don't see the benefit of doing this if all of this stuff is being generated together regardless.